### ISSUE TITLE
False positive for the `no-identical-title` rule

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#120]
This fails as `'foo'` is used twice.

```js
if (process.platform === 'win32') {
	test('foo', t => {
		// Windows specific testing
	});
} else {
	test('foo', t => {

	});
}
```

We should not warn when either of the tests are unreachable. Maybe a good case for code path analysis?

**Comment by jfmengels** [comment]
Hmm. I understand the concern, but that would seem like a valid case to change the title name IMO. It's a different test, and should it fail only on some platforms because it is platform specific, then it'd be nice to have a unique title for it (or you could look at the wrong test for a while before noticing you're not looking at the right one).

That said, I'm not against doing this if someone wants to have a go at this. And this is definitely a case for code path analysis.

**Comment by sindresorhus** [comment]
Valid points. Let's see what others think.

**Comment by jamestalmage** [comment]
I agree with @jfmengels.

In your example you should rename the tests `foo - windows` and `foo - unix/OSX`

**Comment by sindresorhus** [comment]
👍 Cool
